source/extensions/dynamic_modules/sdk/rust/src/lib.rs

```diff
@@ -1113,18 +1125,20 @@ impl EnvoyHttpFilter for EnvoyHttpFilterImpl {
     }
   }
 
-  fn get_most_specific_route_config(&self) -> Option<&'static dyn Any> {
+  fn get_most_specific_route_config(&self) -> Option<std::sync::Arc<dyn Any>> {
```

Comment:
using arc wrapping this any doesn't solve the underlying problem; which is that this reference counting doesn't prevent underlying envoy object gets freed. you don't need this by simply having the "inner arc" in your own PerRouteFilter struct.